src/InvestmentManager.sol

```diff
@@ -352,6 +343,8 @@ contract InvestmentManager is Auth {
         state.maxWithdraw = state.maxWithdraw + currencyPayout;
         state.pendingDepositRequest = remainingInvestOrder;
 
+        if (remainingInvestOrder == 0) state.pendingCancelDepositRequest = false;
```

Comment:
I think we have to set ` state.pendingCancelDepositRequest = false;`  also in the `handleExecutedCollectInvest` function. Maybe the order got already fulfilled and `remainingInvestOrder=0` and there is nothing to cancel anymore... same for `handleExecutedCollectInvest`